File: platforms/kotlin/examples/core/texture_mip_chain/prepare.kt
import org.fragmentcolor.*

// Encoded path: pass bytes plus the format you expect.
val encoded_png_bytes: ByteArray = byteArrayOf()
val chain = TextureMipChain.prepare(encoded_png_bytes, TextureFormat.RGBA8_UNORM_SRGB, null)

// Raw path: include the size so prepare skips decoding.
val raw_rgba: ByteArray = ByteArray(8 * 8 * 4)
val chain_raw = TextureMipChain.prepare(raw_rgba, TextureFormat.RGBA8_UNORM_SRGB, Size(width=8u, height=8u, depth=null))

// Upload the chain through the regular create_texture entry point.
val renderer = Renderer()
val texture = renderer.createTexture(TextureInputMobile.Prepared(chain), null)

File: platforms/swift/examples/core/texture_mip_chain/prepare.swift
import FragmentColor
import Foundation

// Raw RGBA path: include the size so prepare skips decoding.
let rawRgba = Data(repeating: 200, count: 8 * 8 * 4)
let chainRaw = try TextureMipChain.prepare(
    bytes: rawRgba,
    format: .rgba8UnormSrgb,
    size: Size(width: 8, height: 8, depth: nil)
)

// Upload the chain through the regular createTexture entry point.
let renderer = Renderer()
let texture = try await renderer.createTexture(input: .prepared(chainRaw))
let _ = chainRaw.levelCount()
let __ = texture.size()